### PR TITLE
Static analysis: Fix usage of XmlReader to ensure XmlResolver is null

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
The default XmlResolver will attempt to resolve external resources, recommendation is to explicitly set to null which will raise exception if malicious xml attempts to cause xmlreader to access external resources.  Dotnet Core 2.0 now has XmlResolver.

Static analysis bugs:

TFS:12477148, 12477346, 12477347, 12477929, 12477938, 12477942, 12478520, 12478531, 12478540, 12478800, 12478801, 12478803, 12478810, 12478811, 12479280, 12479288, 12479295, 12479302, 12479304

### CODE REVIEW

File: src/System.Management.Automation/engine/serialization.cs [hunk 1]
```diff
@@ -4051,14 +4051,14 @@ private static XmlReaderSettings GetXmlReaderSettingsForUntrustedXmlDocument()
             settings.IgnoreWhitespace = true;
             settings.MaxCharactersFromEntities = 1024;
             settings.MaxCharactersInDocument = 512 * 1024 * 1024; // 512M characters = 1GB
+            settings.XmlResolver = null;
 
 #if CORECLR // DtdProcessing.Parse Not In CoreCLR
             settings.DtdProcessing = DtdProcessing.Ignore;
 #else       // XmlReaderSettings.ValidationFlags/ValidationType/XmlResolver Not In CoreCLR
```
Comment (mirichmo) [inline src/System.Management.Automation/engine/serialization.cs]: Did they bring back the rest of these as well?

Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/serialization.cs]: Good question, let me check

Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/serialization.cs]: Looks like they are back, updated code

File: src/System.Management.Automation/engine/serialization.cs [hunk 2]
```diff
@@ -4026,14 +4026,14 @@ private static XmlReaderSettings GetXmlReaderSettingsForCliXml()
             xrs.IgnoreProcessingInstructions = true;
             xrs.IgnoreWhitespace = false;
             xrs.MaxCharactersFromEntities = 1024;
+            xrs.XmlResolver = null;
             //xrs.DtdProcessing = DtdProcessing.Prohibit; //because system.management.automation needs to build as 2.0
             //xrs.ProhibitDtd = true;
 #if !CORECLR
```
Comment (mirichmo) [inline src/System.Management.Automation/engine/serialization.cs]: Here too

Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/serialization.cs]: will fix

**Comment by SteveL-MSFT** [comment]
@mirichmo comments addressed, can you merge?

**Review by mirichmo — APPROVED** [review]

**Pull request merged**